manager.go

```diff
@@ -587,7 +625,7 @@ func (mgr *Manager) LoadDataDir() error {
 					continue
 				}
 				// we have already validated the pindex paths, hence feeding directly
-				pindex, err := OpenPIndex(mgr, req.path)
+				pindex, err := mgr.OpenPIndexBasedOnStatus(req.path)
```

Comment:
this was an exported method which shouldn't be changed. 
You could call a new wrapper method that could reuse the existing APIs along with additional control logic